optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/OptaShiftRosteringEntryPoint.java

```diff
@@ -45,4 +72,13 @@ public void onTenantsReady(final @Observes TenantStore.TenantsReady tenantsReady
             DomGlobal.document.body.appendChild(navigationController.getAppElement());
         }));
     }
-}
\ No newline at end of file
+
+    private void healthCheck() {
+        TenantRestServiceBuilder.getTenantList(FailureShownRestCallback.onSuccess(tenantList -> {
+            if (null == tenantList) {
+                throw new IllegalStateException("The server cannot be contacted on url (" + Window.Location.getHref() + ").");
+            }
+            tenantStore.init(); //FIXME: Shouldn't this call be made by the Container once it's annotated with @PostConstruct?
+        }));
+    }
```

Comment:
I'm not sure it's optimal do use this endpoint as a health-checker. Also, we'll have a duplicated call for the tenant list because `tenantStore.init()` does that too. I think it's okay to merge this but maybe adding a FIXME or a TODO marker would be a good idea.

Reply:
It might be a good idea to create an endpoint specificity for health checks. All other endpoints would require advance knowledge of existing tenants (this is the only one with no parameters).